### ISSUE TITLE
Characters drop above 19200

### ISSUE THREAD

**Issue by ubidefeo** [mladenilic/arduino-cereal#1]
hi

I found this through my team-mate Roberto.
I wanted to test it to see if we could use it in tutorials for our CLI.
I'm testing with one of my sketches which communicates over Serial with a Python script at 115200 and have a lot of dropped characters so I started at 9600 and went gradually up. 19200 was fine but at 38400 it was failing again.

Interesting project, I'll test the library next 👍

**Comment by mladenilic** [comment]
Hi @ubidefeo,

First of all, thanks a lot for taking the interest in Cereal!

I couldn't recreate the issue you are having, one possibility is that you started your Cereal monitor with wrong baud settings. Please check the following image and make sure that your baud settings on the Arduino matches the baud rate of your monitor.

![baud](https://user-images.githubusercontent.com/7901221/99855154-73012900-2b86-11eb-86a8-4f61d81592b3.png)

You can set the baud rate either using command line argument like this: `arduino-cereal --baud=38400`. Your other option is to use `cereal.config.json` as described in the readme. Status bar on the top, always shows your current setting.

Here's also code that I used to generate the image above, so you can try and replicate my results:

```cpp
int i = 0;

void setup() {
  Serial.begin(38400);  
}

void loop() {
  Serial.println("line: " + String(i++, DEC));
}
```

**Note:** I did notice an unrelated bug, where in certain cases new line character get's eaten up and two lines are printed together on same line. I am looking into fixing this.

**Comment by ubidefeo** [comment]
@mladenilic 
I have of course launched it properly and kept the same baudrate in the sketch, otherwise I'd have the weirdest output 😁 
New line characters seem to be the ones which get dropped often at speeds higher than 19200 (on my MBP15 i9 32GB), but I had one instance of a line completely disappearing.

Happy I might have helped you finding the bug.
I'll watch the repo, so I can follow the development 😃

**Comment by mladenilic** [comment]
Hi @ubidefeo,

I've released `v1.0.3`, fixing the issues with monitor sometimes eating up or doubling line breaks. You can update with `npm update -g arduino-cereal`.

Please try again and let me know if this fixes the issue. If you are still getting issues with output, couple you please share a code snippet that I could you to recreate the issue.

Thanks! :)

**Comment by ubidefeo** [comment]
hey @mladenilic 
it works great at 115200 now :)
I push a lot of data through and it didn't flinch, only every now and then some characters overlap the bottom line.
<img width="1988" alt="Screenshot 2020-11-22 at 22 08 13" src="https://user-images.githubusercontent.com/1894487/99917255-5e03d180-2d0f-11eb-93ee-2137e34c50e7.png">

Any chance you can add a setting for the column split?
I'd love to have something like percentage for the left column which would also support `0` in order to make it disappear and `100` to remove the right column :)

Thanks for the fast iteration
ubi

**Comment by mladenilic** [comment]
Hey @ubidefeo,

Glad to hear it worked! I haven't had a chance to look into the issue with the long, single line outputs. I'll open new issue to track that one.

I have released v1.1. It includes an input panel, to allow sending data to the board (Still have to update readme). This release also includes options to disable variables or input panel. You can do this using config file.

```json5
{
 // ...

  "theme": {

    "modules": {
      "variables": {
        "enabled": false
      },

      "input": {
        "enabled": false,
      }
    }
  }
}

```
I wasn't sure about allowing users to control width of the variables panel, reducing the width of it below the default value would most probably cause the layout to break in case variable names are too long or range variables are used.

Also, there is no option to disable monitor panel. Let me know if you see a use case where disabling a monitor panel would make sense.

Thanks!

**Comment by ubidefeo** [comment]
nice tip :)
I'd disable the monitor in case I only want to use the lib to watch variables,
but to be honest I see hiding the first panel more useful than the second :)

nice job
u.
